### DISCUSSION TITLE
Way to auto generate swagger on Adonis V6 ?

### DISCUSSION THREAD

**Discussion by Erwann-Dev** [adonisjs/core#4714] [Help]
Is there a way to auto-generate Swagger documentation for APIs in AdonisJS V6? If so, what tools or techniques can be used to integrate it effectively?

**Comment by shemzz** [comment]
I used this and it works fine
[adonis autoswagger](https://github.com/ad-on-is/adonis-autoswagger/tree/main)

My only issue so far is getting it to work in Docker. give it a try

**Reply by Kishan2029** [reply]
Can i create swagger doc on Adonis V6 using [adonis5-swagger](https://www.npmjs.com/package/adonis5-swagger) package >

**Reply by shemzz** [reply]
it will be better to use  [auto swagger](https://github.com/ad-on-is/adonis-autoswagger/tree/main) for adonis v6 projects

**Comment by Kishan2029** [comment]
I am using [auto swagger](https://github.com/ad-on-is/adonis-autoswagger/tree/main) in the adonis V6. By default there are 3 schemas for global authorisation. Below is the yml representation.
` securitySchemes:
    BearerAuth:
      type: "http"
      scheme: "bearer"
    BasicAuth:
      type: "http"
      scheme: "basic"
    ApiKeyAuth:
      type: "apiKey"
      in: "header"
      name: "X-API-Key"`
     
 I want to override this and only use ApiKeyAuth with custom header name.  I changed the securitySchemas in the swagger config as mentioned in the [documentation](https://github.com/ad-on-is/adonis-autoswagger?tab=readme-ov-file). It is changing ApiKeyAuth header name but still shows all three authorisation method.

`   securitySchemes: {
    ApiKeyAuth: {
      type: 'apiKey',
      in: 'header',
      name: 'x-auth-apikey', 
    },
  },

  security: [
    {
      ApiKeyAuth: [],
    },
  ],`
  
  Can any one help ?
